clusterloader2/pkg/prometheus/prometheus.go

```diff
@@ -377,11 +452,12 @@ func (pc *Controller) exposeAPIServerMetrics() error {
 		return err
 	}
 	createClusterRoleBinding := func() error {
-		_, err := clientSet.GetClient().RbacV1().ClusterRoleBindings().Create(context.TODO(), &rbacv1.ClusterRoleBinding{
-			ObjectMeta: metav1.ObjectMeta{Name: "system:anonymous"},
+		_, err := testClusterClientSet.RbacV1().ClusterRoleBindings().Create(context.TODO(), &rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{Name: "apiserver-metrics-viewer-binding"},
 			RoleRef:    rbacv1.RoleRef{Kind: "ClusterRole", Name: "apiserver-metrics-viewer"},
```

Comment:
The `apiserver-metrics-viewer` cluster role does not grant permission to access the kubelet metrics, causing the scraping to fail (while it was succeeding when using the prometheus service account). The difference seems to be due to the lack of permissions on `nodes/metrics`, previously granted by the prometheus-k8s cluster role:

```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: prometheus-k8s
rules:
- apiGroups:
  - ""
  resources:
  - nodes/metrics
  - services
  - endpoints
  - pods
  verbs:
  - get
  - list
  - watch
- nonResourceURLs:
  - /metrics
  verbs:
  - get
```

Reply:
Yup, we noticed that yesterday as well and fixed properly today in https://github.com/kubernetes/perf-tests/pull/3240 :).

Sorry for the noise.

Reply:
Oh, that was super quick, I didn't expect the updated version to have already been merged. Thanks!